src/include/OSL/oslexec.h

```diff
@@ -1031,6 +1031,18 @@ class OSLEXECPUBLIC ShadingSystem {
     void add_symlocs(cspan<SymLocationDesc> symlocs);
     void add_symlocs(ShaderGroup* group, cspan<SymLocationDesc> symlocs);
 
+    // Find the SymLocationDesc for this named param, returning its pointer
+    // or nullptr if that name is not found.
+    const SymLocationDesc* find_symloc(ustring name) const;
+    const SymLocationDesc* find_symloc(ShaderGroup* group, ustring name) const;
+
+    // Find the SymLocationDesc for this named param but only if it matches
+    // the arena type, returning its pointer or nullptr if that name is not
+    // found.
+    const SymLocationDesc* find_symloc(ustring name, SymArena arena) const;
+    const SymLocationDesc* find_symloc(ShaderGroup* group, ustring name,
+                                       SymArena arena) const;
+
```

Comment:
These additions are not strictly necessary for the rest of this PR to work, but it did seem to be useful to query current symbol locations for a shader group inside the `build_attribute_getter` function.